### ISSUE TITLE
solc 0.7.4 file-level functions parser error

### ISSUE THREAD

**Issue by gitpusha** [protofire/solhint#262]
The parser used by solhint doesnt seem to support solc 0.7.4 file level functions yet. 

I get this:

![image](https://user-images.githubusercontent.com/36712489/96714201-19001e80-13a2-11eb-89cc-afda523a06c0.png)

For something like this:

```solidity
/* solhint-disable */
pragma solidity 0.7.4;

function add(uint x, uint y) pure returns (uint z) {
    require((z = x + y) >= x, "ds-math-add-overflow");
}
function sub(uint x, uint y) pure returns (uint z) {
    require((z = x - y) <= x, "ds-math-sub-underflow");
}
function mul(uint x, uint y) pure returns (uint z) {
    require(y == 0 || (z = x * y) / y == x, "ds-math-mul-overflow");
}
... 

```

Also adding `/* solhint-disable */` atop didnt stop the linter from linting. 

However, putting said file in `.solhintignore` worked.

**Comment by fvictorio** [comment]
Are you sure? I just tested it with a file with a top-level function (for other reasons) and it worked fine.

**Comment by gitpusha** [comment]
Yep pretty sure: you can try it out [here](https://github.com/gelatodigital/gelato-instadapp/blob/master/contracts/vendor/DSMath.sol). 

Using `"solhint": "3.3.0"`

**Comment by gitpusha** [comment]
Seems to be resolved with version `0.3.1` . Correct?

**Comment by fvictorio** [comment]
Should be fixed in `solhint@3.3.2`, please re-open if it doesn't.
